pkg/translate/transform.go

```diff
@@ -0,0 +1,91 @@
+package translate
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+	"go.expect.digital/translate/pkg/model"
+	translatev1 "go.expect.digital/translate/pkg/pb/translate/v1"
+)
+
+// ----------------------Common types----------------------
+
+// uuidToProto converts uuid.UUID to string.
+func uuidToProto(u uuid.UUID) string {
+	if u == uuid.Nil {
+		return ""
+	}
+
+	return u.String()
+}
+
+// uuidFromProto converts string to uuid.UUID.
+func uuidFromProto(s string) (uuid.UUID, error) {
+	if s == "" {
+		return uuid.Nil, nil
+	}
+
+	id, err := uuid.Parse(s)
+	if err != nil {
+		return uuid.Nil, fmt.Errorf("parse uuid: %w", err)
+	}
+
+	return id, nil
+}
+
+// ----------------------Service----------------------
+
+// serviceToProto converts model.Service to translatev1.Service.
+func serviceToProto(s *model.Service) *translatev1.Service {
+	return &translatev1.Service{Id: uuidToProto(s.ID), Name: s.Name}
+}
+
+// serviceFromProto converts translatev1.Service to model.Service.
+func serviceFromProto(s *translatev1.Service) (*model.Service, error) {
+	var (
+		service = &model.Service{Name: s.Name}
+		err     error
+	)
+
+	service.ID, err = uuidFromProto(s.Id)
+	if err != nil {
+		return nil, fmt.Errorf("transform id: %w", err)
+	}
+
+	return service, nil
+}
+
+// servicesToProto converts []model.Service to []*translatev1.Service.
+func servicesToProto(s []model.Service) []*translatev1.Service {
```

Comment:
Use a generic function to transform a slice. E.g.

```suggestion
func servicesToProto(s []model.Service) []*translatev1.Service { return sliceToProto(s) }

func sliceToProto[T any, R any](slice []T, transform func(*T) *R) []R {
	if len(slice) == 0 {
		return nil
	}

	v := make([]R, 0, len(slice))

	for i := range slice {
		v = append(v, *transform(&slice[i]))
	}

	return v
}
```